### ISSUE TITLE
space isnt treated like enter in tab completion

### ISSUE THREAD

**Issue by ribosomerocker** [nushell/nushell#9227]
### Describe the bug

It's generally expected that space would be treated similarly to accepting the currently selected completion option when it's on, similarly to enter. This makes UX way better for the completion.

### How to reproduce

1. Write anything
2. Press tab
3. Select an option
4. Press space

### Expected behavior

I expected nu to accept the selection-- yet the tab completion menu still persists.

### Configuration

irrelevant

**Comment by ShinyZero0** [comment]
That should be configurable, idk who "generally expects" exactly that behavior. Imo `space` should abort completion and insert a space because when i press `space` i want to insert a space. We already have `enter` for accepting completion. Though how `space` and failed completion currently work is the worst case of how they could work: that "no records" thing is an eyesore, it triggers unwanted completion after space and you have to press `enter` twice to execute a command.

**Comment by amtoine** [comment]
agree with @ShinyZero0, i think i expect `space` to actually insert a space :thinking:

**Comment by fdncred** [comment]
I'm not sure about this since space can be part of command names. I generally expect space to put in a space character. If it canceled things we'd have a real problem completing commands with spaces in them I think.

**Comment by ShinyZero0** [comment]
It completes if i type the command, abort completion and press tab like this:

```bash
com<tab><esc> # complete and close menu
command <tab> # menu with subcommands
```
That's if you have no vi mode, in vi mode that'd be `com<tab><backspace><space>` or with `escape` and then `a`

**Comment by ShinyZero0** [comment]
Imo there's no problem with subcommands as probably when you complete the main command you don't even want to see all the subcommands. Pressing tab one more time makes sense there, and makes much more than these backspace tricks to stop nu from causing epilepsia stroke with blinking filenames, annoying `NO RECORDS FOUND` and accidentally moving my shell to top of the screen to free space for these blinking filenames after i just press tab to check if nu can complete just one flag

**Comment by ShinyZero0** [comment]
I almost forgot that failed completion also causes lags before every types character if you don't abort it. Related #9045

**Comment by rsteube** [comment]
Yes, lots of arguments have space in it so using it to accept it isn't exactly a good idea.
But I think this is coming from the point of view that nushell is currently still missing a modal completion menu with filtering as pointed out by @ShinyZero0 in the other issue and doing this inline.
So here it could make sense to force the use of escaping (`\ `) to enter space as the actual command line is edited (unless quotes are used).
There are certainly different preferences but i would highly recommend having a look how this is done in elvish.
The horizontal arrangement in columns makes moving between them a lot faster and the modal menu provides filtering without reinvoking the completion (which can be quite slow).

**Comment by ribosomerocker** [comment]
> agree with @ShinyZero0, i think i expect `space` to actually insert a space thinking

My suggestion doesn't elide this. If my suggestion is accepted, pressing space would both accept the completion and insert a space character.

Also, I've just tried something else, and it seems when i keep pressing `tab`, it doesn't continue to choose a different entry in the tab completion. That's another bug that needs fixing, in my opinion.

**Comment by ShinyZero0** [comment]
> pressing space would both accept the completion and insert a space character
I can do it with enter. 
> when i keep pressing tab, it doesn't continue to choose a different entry
For me it works, but it doesn't when i break my config. Maybe something with your config?

**Comment by ribosomerocker** [comment]
> I can do it with enter.
I know. I have already mentioned this in the original issue, but I usually do it with space using ZSH and I think that it would be an improvement to have this.
